<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>981</width>
      <height>461</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_1">
     <attribute name="title">
      <string>Interpolation</string>
     </attribute>
     <widget class="QCustomPlot" name="interPlot" native="true">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>10</y>
        <width>691</width>
        <height>411</height>
       </rect>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>261</width>
        <height>371</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox {
  border: 1px solid gray;
}</string>
      </property>
      <property name="title">
       <string>Methods</string>
      </property>
      <widget class="QRadioButton" name="radioButtonNewton">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>20</y>
         <width>161</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Newton polynomial</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="radioButtonSpline">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>161</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Cubic Spline</string>
       </property>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>140</y>
         <width>241</width>
         <height>31</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButtonDrawPlot">
          <property name="styleSheet">
           <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}</string>
          </property>
          <property name="text">
           <string>Draw plot</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>70</y>
         <width>239</width>
         <height>61</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinBoxNumPoints">
          <property name="minimum">
           <number>100</number>
          </property>
          <property name="maximum">
           <number>9000</number>
          </property>
          <property name="value">
           <number>300</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_1">
          <property name="text">
           <string> Number of points: </string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string> Degree of polynominal: </string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBoxDegreePoly">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>9</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="Line" name="line">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>241</width>
         <height>16</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_9">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>330</y>
         <width>241</width>
         <height>31</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_10">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButtonCalculate">
          <property name="styleSheet">
           <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}</string>
          </property>
          <property name="text">
           <string>Calculate</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>200</y>
         <width>240</width>
         <height>121</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="1" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string> Result:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDateTimeEdit" name="dateTimeEdit">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="dateTime">
           <datetime>
            <hour>0</hour>
            <minute>0</minute>
            <second>0</second>
            <year>2021</year>
            <month>1</month>
            <day>1</day>
           </datetime>
          </property>
          <property name="maximumDate">
           <date>
            <year>2030</year>
            <month>12</month>
            <day>31</day>
           </date>
          </property>
          <property name="displayFormat">
           <string>yyyy-MM-dd HH:mm</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string> Newton:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string> Enter date: </string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string> Cubic Spline:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="lineEditResultNewton">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="lineEditResultSpline">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Approximation</string>
     </attribute>
     <widget class="QCustomPlot" name="approxPlot" native="true">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>10</y>
        <width>691</width>
        <height>411</height>
       </rect>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox2_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>261</width>
        <height>301</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox {
  border: 1px solid gray;
}</string>
      </property>
      <property name="title">
       <string>Least squares method</string>
      </property>
      <widget class="QWidget" name="gridLayoutWidget_7">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>120</y>
         <width>241</width>
         <height>31</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButtonDrawPlot_a">
          <property name="styleSheet">
           <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}</string>
          </property>
          <property name="text">
           <string>Draw plot</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>239</width>
         <height>81</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_8">
        <item row="1" column="1">
         <widget class="QSpinBox" name="spinBoxDegreePoly_a">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>20</number>
          </property>
          <property name="value">
           <number>3</number>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string> Degree of polynominal: </string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinBoxNumPoints_a">
          <property name="minimum">
           <number>100</number>
          </property>
          <property name="maximum">
           <number>9000</number>
          </property>
          <property name="value">
           <number>300</number>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string> Number of points: </string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string> Extend the graph (days):</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="spinBoxDaysExt">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>365</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="Line" name="line_a">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>160</y>
         <width>241</width>
         <height>16</height>
        </rect>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_10">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>260</y>
         <width>241</width>
         <height>31</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_11">
        <item row="0" column="0">
         <widget class="QPushButton" name="pushButtonCalculate_a">
          <property name="styleSheet">
           <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}</string>
          </property>
          <property name="text">
           <string>Calculate</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="gridLayoutWidget_8">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>240</width>
         <height>71</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_9">
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineEditResultApprox">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string> Result:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string> Enter date: </string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDateTimeEdit" name="dateTimeEdit_a">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="dateTime">
           <datetime>
            <hour>0</hour>
            <minute>0</minute>
            <second>0</second>
            <year>2021</year>
            <month>1</month>
            <day>1</day>
           </datetime>
          </property>
          <property name="maximumDate">
           <date>
            <year>2030</year>
            <month>12</month>
            <day>31</day>
           </date>
          </property>
          <property name="displayFormat">
           <string>yyyy-MM-dd HH:mm</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>480</y>
      <width>981</width>
      <height>151</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QGroupBox {
  border: 1px solid rgb(160, 160, 160);
  border-radius: 3px;
}</string>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QTextEdit" name="textInfo">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>40</y>
       <width>691</width>
       <height>101</height>
      </rect>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QWidget" name="horizontalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>10</y>
       <width>691</width>
       <height>31</height>
      </rect>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="pushButtonInfo">
        <property name="styleSheet">
         <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}
</string>
        </property>
        <property name="text">
         <string>Info</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonClear">
        <property name="styleSheet">
         <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}</string>
        </property>
        <property name="text">
         <string>Clear Info</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
    <widget class="QGroupBox" name="groupBox1">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>261</width>
       <height>131</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QGroupBox {
  border: 1px solid gray;
}</string>
     </property>
     <property name="title">
      <string>Data</string>
     </property>
     <widget class="QWidget" name="gridLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>241</width>
        <height>101</height>
       </rect>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButtonOpen">
         <property name="styleSheet">
          <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}
</string>
         </property>
         <property name="text">
          <string>Open CSV</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="pushButtonCleanData">
         <property name="styleSheet">
          <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}
</string>
         </property>
         <property name="text">
          <string>Clean Data</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="pushButtonReset">
         <property name="styleSheet">
          <string notr="true">QPushButton {
  background-color: rgb(80, 80, 80);
  color: white;
  border-radius: 5px;
  border: 1px solid gray;
}
</string>
         </property>
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
